### ISSUE TITLE
Add a subject for Questions without using an Annotation

### ISSUE THREAD

**Issue by miles-mason** [serenity-bdd/serenity-core#1494]
It's extremely useful to add Subjects to Serenity Screenplay Questions. However, the current approach using the `@Subject` annotation is very restrictive, since it needs to be placed on the exact class or method, and any arguments to it need to be compile time constants.

It would be useful to have a more flexible way to supply subjects, for example implementing a method in the `Question` Interface itself.

**Comment by wakaleo** [comment]
We've considered this. However, adding another method to the Question interface would make it impossible to declare a question as a lambda, which is a very useful pattern. The compromise solution is to have an extra parameter in the `seeThat()` method, e.g.

```
        sam.should(
                seeThat("the user name", TheUser.nameForUserWithId(1L), name -> name.equalsIgnoreCase("George"))
        );
```

There isn't a way in Java as far as I know to do any better than this.

**Comment by miles-mason** [comment]
The compromise solution is quite poor because then the `Subject` isn't attached to the `Question` itself, but to a single use of the `Question`, potentially making for duplication of String literals and maintenance difficulty. Could there not be another interface like `QuestionWithSubject` that could be optionally used?

**Comment by miles-mason** [comment]
This was resolved in dd002df974e2d0d5f4f5f34d2f0c30ee67a9b742

Thanks @wakaleo !
